app/src/main/java/com/keylesspalace/tusky/TuskyApplication.kt

```diff
@@ -69,10 +70,10 @@ class TuskyApplication : Application(), HasAndroidInjector {
         AppInjector.init(this)
 
         // Migrate shared preference keys and defaults from version to version. The last
-        // version that did not have a SCHEMA_VERSION was 97, so that's the default.
-        val oldVersion = sharedPreferences.getInt(PrefKeys.SCHEMA_VERSION, 97)
-        if (oldVersion != BuildConfig.VERSION_CODE) {
-            upgradeSharedPreferences(oldVersion, BuildConfig.VERSION_CODE)
+        // version that did not have a SCHEMA_VERSION was 100, so that's the default.
+        val oldVersion = sharedPreferences.getInt(PrefKeys.SCHEMA_VERSION, 100)
+        if (oldVersion != SCHEMA_VERSION) {
```

Comment:
Does this support a step-by-step upgrade? (So a full range of versions inbetween can be enumerated.)

Otherwise you wouldn't be able to do more than one upgrade in total.

(See https://github.com/tuskyapp/Tusky/pull/3205)

Reply:
After thinking about this for a bit: I think I like it.

Especially the expressiveness of the number (giving the date/an age without further effort).

I would get rid of the counter though for readability.

(It is highly unlikely to have two at the same day. And/or on fixing the inevitable merge conflict it is probably already a day later.)

Reply:
https://wiki.lspace.org/Million-to-one_chance :-)  I think trading off two extra digits is worthwhile.